src/Catel.Analyzers/Supressors/IDE0051OnPropertyChangeSupressor.cs

```diff
@@ -0,0 +1,186 @@
+                                                    where descendantNode is AttributeSyntax && string.Equals(descendantNode.GetIdentifier(), CatelFodyAttributeLookupName)
+                                                    select descendantNode.FirstAncestor<PropertyDeclarationSyntax>();
+
+                    if (IsAnyExposedPropertyDeclarationMatch(exposedMarkedDeclarations, exposeAttributeType, propertyName, rootSemantic, context.CancellationToken))
+                    {
+                        context.ReportSuppression(Suppression.Create(supresssionDescriptor, diagnostic));
+                    }
+
+                    if (context.CancellationToken.IsCancellationRequested)
+                    {
+                        return;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
+        }
+
+        private bool IsAnyExposedPropertyDeclarationMatch(IEnumerable<PropertyDeclarationSyntax> propertyDeclarations, INamedTypeSymbol attributeTypeToMatch, string propertyName, SemanticModel model, CancellationToken cancellationToken)
+        {
+            foreach (var property in propertyDeclarations.Distinct())
+            {
+                var propertySymbol = model.GetDeclaredSymbol(property, cancellationToken);
+
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    return false;
+                }
+
+                var exposeAttributesList = propertySymbol.GetAttributes().Where(x => SymbolEqualityComparer.Default.Equals(x.AttributeClass, attributeTypeToMatch)).ToList();
+
+                foreach (var exposeAttribute in exposeAttributesList)
+                {
+                    if (exposeAttribute.ConstructorArguments.Any())
+                    {
+                        var constructorArgValue = exposeAttribute.ConstructorArguments.FirstOrDefault().Value.ToString();
+
+                        if (string.Equals(constructorArgValue, propertyName))
+                        {
+                            return true;
+                        }
+                    }
+
+                    if (exposeAttribute.NamedArguments.Any())
+                    {
+                        var argument = exposeAttribute.NamedArguments.FirstOrDefault(arg => string.Equals(arg.Key, "propertyName"));
+
+                        var propertyIsExposed = argument.Equals(default(KeyValuePair<string, TypedConstant>)) ? false : string.Equals(argument.Value.Value.ToString(), propertyName);
+
+                        if (propertyIsExposed)
+                        {
+                            return true;
+                        }
+                    }
+                }
+
+                if (cancellationToken.IsCancellationRequested)
```

Comment:
This one is not really needed. It's either the end of the for-loop or it will be checked in the next loop.